### PR TITLE
Remove long-deprecated HL_TRACE env var support

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

**Comment by steven-johnson** [comment]
(Note: I'm tempted to also add a target-string substitution that will allow `trace_all` as a shortcut for `trace_loads-trace_stores-trace_realizations`, but I'm a little unsure as to whether allowing synonyms in target string parsing/generation would open a possible rathole of badness. Thoughts?)

File: src/Tracing.cpp [hunk 1]
```diff
@@ -41,25 +41,13 @@ struct TraceEventBuilder {
...
-                         << "trace_stores, and trace_realizations instead\n";
-            int l = std::stoi(global_level);
-            trace_all_loads |= l > 2;
-            trace_all_stores |= l > 1;
-            trace_all_realizations |= l > 0;
-        }
+        : env(e),
+          trace_all_loads(t.has_feature(Target::TraceLoads)),
+          trace_all_stores(t.has_feature(Target::TraceStores)),
+          trace_all_realizations(t.has_feature(Target::TraceRealizations)) {
```
Comment (abadams) [inline src/Tracing.cpp]: As a drive-by-fix could I ask you to set trace_all_realizations to true when trace_loads or trace_stores is on too? They don't work without trace_all_realizations being on (and the errors are missing symbol mysterious nonsense).

**Comment by abadams** [comment]
I was only proposing adding new overheads for realization-into-existing buffers. I don't care about the other ones default constructing a ParamMap. They're slow anyway.

**Comment by steven-johnson** [comment]
> I was only proposing adding new overloads 

I think you added this to the wrong PR :-)

File: src/Tracing.cpp (hunk 1 above)
Comment (steven-johnson) [reply src/Tracing.cpp]: Would you ever want to trace realizations without loads or stores? (ie should it even be a separate flag?)

Comment (steven-johnson) [reply src/Tracing.cpp]: (done)

**Comment by abadams** [comment]
So I did.

**Comment by abadams** [comment]
To answer your question: Because you might want to check what granularity things are being computed at. E.g. how large is the intermediate buffer for something.
